homework3/semi-uber/Server/api/routes/registration.router.js

```diff
@@ -1,40 +1,47 @@
 const express = require("express");
 const router = express.Router();
+const config = require("config");
+const { secret } = config.get("JWT");
+const saltRounds = Number(config.get("saltRounds"));
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const User = require("../Schemas/user.schema");
 const writeLog = require("../middlewars/writeLog");
-const mongoose = require("mongoose");
-const User = mongoose.model(
-  "User",
-  mongoose.Schema({
-    username: String,
-    password: String,
-    email: String,
-    status: String
-  })
-);
 
-const saveUser = user => {
-  user
-    .save()
-    .then(res.json(user))
-    .catch(err => {
-      console.error(err.name);
-      throw err;
-    });
+const hashPassword = (password, saltRounds) => {
+  return bcrypt.hash(password, saltRounds);
+};
+
+const createToken = (body, secret) => {
+  return { token: jwt.sign(body, secret), password: body.password };
 };
 
 router.post("/", writeLog, (req, res) => {
   const { username, password, email, status } = req.body;
-  const user = new User({
-    username,
-    password,
-    email,
-    status
-  });
 
-  User.find({ username }).then(answer =>
-    answer.length === 0
-      ? saveUser(user)
-      : res.json({ status: "Nickname is allready taken" })
-  );
+  hashPassword(password, saltRounds)
+    .then(password => createToken({ username, password }, secret))
+    .then(
+      res =>
+        new User({
```

Comment:
I would suggest you to start using `Joi` already, now is the best moment to add it to your project.

Reply:
Done